backend/src/main/kotlin/code/nebula/cipherquest/handler/GlobalExceptionHandler.kt

```diff
@@ -1,14 +1,22 @@
 package code.nebula.cipherquest.handler
 
 import code.nebula.cipherquest.exceptions.DocumentNotFoundException
+import org.springframework.dao.DataIntegrityViolationException
 import org.springframework.http.HttpStatus
 import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.ResponseStatus
 import org.springframework.web.bind.annotation.RestControllerAdvice
 
 @RestControllerAdvice
 class GlobalExceptionHandler {
     @ExceptionHandler(DocumentNotFoundException::class)
     fun handleDocumentNotFoundException(e: DocumentNotFoundException): ResponseEntity<String> =
         ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.message)
+
+    @ExceptionHandler(DataIntegrityViolationException::class)
```

Comment:
This method will catch all data integrity violation exceptions, we should have a custom exception or return a generic error message by removing the Elvis operator.